File: Day 05/485_MaxConsecutiveOnes.cpp
// 485. Max Consecutive Ones
// Easy
// Topics
// Companies
// Hint
// Given a binary array nums, return the maximum number of consecutive 1's in the array.
// Example 1:
// Input: nums = [1,1,0,1,1,1]
// Output: 3
// Explanation: The first two digits or the last three digits are consecutive 1s. The maximum number of consecutive 1s is 3.
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count = 0;
        int maxi=INT_MIN;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==1){
                count++;
                
            }
            else{
               count=0; 
            }
             maxi = max(count,maxi);
        }
        return maxi;      
    }
};

File: Day 40/485_MaxConsecutiveOnes.java
// Given a binary array nums, return the maximum number of consecutive 1's in the array
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int count = 0;
        int maxi = Integer.MIN_VALUE;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 1) {
                count++;
            } else {
                count = 0;
            }
            maxi = Math.max(count, maxi);
        }
        return maxi;
    }
}
